### PR TITLE
Enable more lints used in sdk analysis packages

### PR DESCRIPTION
**Pull request by parlough**
To ease transition to the SDK and for consistency, adds a few linter rules enabled on the [analyzer](https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/analysis_options.yaml) and the [analysis_server](https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/analysis_options.yaml).

The only lint that triggered is `avoid_dynamic_calls`, so this PR fixes those.

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -21,13 +22,18 @@ linter:
     - avoid_returning_this
     - avoid_setters_without_getters
     - avoid_slow_async_io
+    - avoid_unused_constructor_parameters
     - cancel_subscriptions
     - cast_nullable_to_non_nullable
+    - collection_methods_unrelated_type
     - comment_references
     - dangling_library_doc_comments
+    - depend_on_referenced_packages
```
Comment (pq) [inline analysis_options.yaml:31]: This one is already included in core so can be safely removed I think.

**Review by pq — COMMENTED** [review]
What's the analysis you're aligning this with? (Not that I object necessarily.)

Is there a shared set we can include that's a superset of `recommended` that brings these additional analyses in?

**Comment by parlough** [comment]
> What's the analysis you're aligning this with? (Not that I object necessarily.)
> 
> Is there a shared set we can include that's a superset of `recommended` that brings these additional analyses in?

I was just basing this off of the [analyzer](https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/analysis_options.yaml) and the [analysis_server](https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/analysis_options.yaml).

However, I'd be more than happy to switch to [`dart_flutter_team_lints`](https://github.com/dart-lang/ecosystem/blob/main/pkgs/dart_flutter_team_lints/lib/analysis_options.yaml) which includes many of these. It's available in the DEPS file now: https://github.com/dart-lang/sdk/blob/main/DEPS#L142

File: analysis_options.yaml (hunk 1 above)
Comment (parlough) [reply analysis_options.yaml]: Nice catch.

**Comment by pq** [comment]
> I was just basing this off of the [analyzer](https://github.com/dart-lang/sdk/blob/main/pkg/analyzer/analysis_options.yaml) and the [analysis_server](https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/analysis_options.yaml).

OK, this is great. This seems reasonable as we prep for the imminent move to the SDK.  I *do* very much think we should have a shared set of options there and will (re)-kick off that conversation.

Until then let's just land these in-line.

Thanks!

**Review by pq — APPROVED** [review]
Thanks!

**Pull request merged**